x/liquidstakeibc/keeper/msg_server.go

```diff
@@ -381,6 +391,175 @@ func (k msgServer) LiquidStake(
 	return &types.MsgLiquidStakeResponse{}, nil
 }
 
+// LiquidStakeLSM defines a method for liquid staking tokens using the LSM
+func (k msgServer) LiquidStakeLSM(
+	goCtx context.Context,
+	msg *types.MsgLiquidStakeLSM,
+) (*types.MsgLiquidStakeLSMResponse, error) {
+	ctx := sdktypes.UnwrapSDKContext(goCtx)
+
+	for _, delegation := range msg.Delegations {
+		// check if the ibc denom contains the ibc prefix
+		if !strings.HasPrefix(delegation.Denom, types.IBCPrefix) {
+			return nil,
+				errorsmod.Wrapf(
+					types.ErrInvalidLSMDenom,
+					"IBC denom %s doesn't belong to a LSM token",
+					delegation.Denom,
+				)
+		}
+
+		// parse the ibc denom to extract the original LSM token denom
+		hexHash := delegation.Denom[len(types.IBCPrefix):]
+		hexBytes, err := transfertypes.ParseHexHash(hexHash)
+		if err != nil {
+			return nil, errorsmod.Wrapf(err, "could not parse ibc hash from ibc hex %s", hexHash)
+		}
+
+		// get the denom trace from the parsed ibc denom hex hash
+		denomTrace, found := k.ibcTransferKeeper.GetDenomTrace(ctx, hexBytes)
+		if !found {
+			return nil,
+				errorsmod.Wrapf(
+					types.ErrInvalidLSMDenom,
+					"IBC denom %s doesn't belong to a LSM token",
+					delegation.Denom,
+				)
+		}
+
+		// retrieve the host chain associated with the liquid stake action
+		channelId, found := strings.CutPrefix(denomTrace.Path, fmt.Sprintf("%s/", transfertypes.PortID))
```

Comment:
i ibctransfer module usually has functions for all types of string manipulations for ibc prefixes.

Reply:
couldn't find them within the module

Reply:
yup, they are lowercase funcs, ^ this function should take care of.. path which are  `channel1/port1/channel2/port2/channel3/port3`

Reply:
but the path should always be transfer/channel-X isn't it ?